code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,257 @@
+	if(!parallax_initialized || C.updating_parallax) return
+
+	for(var/turf/T in range(get_turf(C.eye),C.view))
+		if(istype(T,/turf/space))
+			C.updating_parallax = 1
+			break
+
+	if(!C.updating_parallax)
+		return
+
+	//multiple sub-procs for profiling purposes
+	if(update_parallax1())
+		update_parallax2(0)
+		update_parallax3()
+		C.updating_parallax = 0
+	else
+		C.updating_parallax = 0
+
+/datum/hud/proc/update_parallax_and_dust()
+	var/client/C = mymob.client
+	if(!parallax_initialized || C.updating_parallax) return
+	C.updating_parallax = 1
+	if(update_parallax1())
+		update_parallax2(1)
+		update_parallax3()
+		C.updating_parallax = 0
+	else
+		C.updating_parallax = 0
+
+/datum/hud/proc/update_parallax1()
+	var/client/C = mymob.client
+	//DO WE UPDATE PARALLAX
+	if(C.prefs.space_parallax)//have to exclude Centcom so parallax doens't appear during hyperspace
+		parallax_on_clients |= C
+	else
+		for(var/obj/screen/parallax/bgobj in C.parallax)
+			C.screen -= bgobj
+		parallax_on_clients -= C
+		C.screen -= C.parallax_master
+		C.screen -= C.parallax_dustmaster
+		return 0
+
+	if(!C.parallax_master)
+		C.parallax_master = getFromPool(/obj/screen/plane_master/parallax_master)
+	if(!C.parallax_dustmaster)
+		C.parallax_dustmaster = getFromPool(/obj/screen/plane_master/parallax_dustmaster)
+	return 1
+
+/datum/hud/proc/update_parallax2(forcerecalibrate = 0)
+	var/client/C = mymob.client
+	//DO WE HAVE TO REPLACE ALL THE LAYERS
+
+	if(!C.parallax.len)
+		for(var/obj/screen/parallax/bgobj in parallax_icon)
+			var/obj/screen/parallax/parallax_layer = getFromPool(/obj/screen/parallax)
+			parallax_layer.appearance = bgobj.appearance
+			parallax_layer.base_offset_x = bgobj.base_offset_x
+			parallax_layer.base_offset_y = bgobj.base_offset_y
+			parallax_layer.parallax_speed = bgobj.parallax_speed
+			C.parallax += parallax_layer
+
+	var/parallax_loaded = 0
+	for(var/obj/screen/S in C.screen)
+		if(istype(S,/obj/screen/parallax))
+			parallax_loaded = 1
+			break
+
+	if(forcerecalibrate || !parallax_loaded)
+		for(var/obj/screen/parallax/bgobj in C.parallax)
+			C.screen |= bgobj
+
+		C.screen |= C.parallax_master
+		C.screen |= C.parallax_dustmaster
+		C.parallax_dustmaster.color = list(0,0,0,0)
+		if(C.prefs.space_dust)
+			C.parallax_dustmaster.color = list(
+			1,0,0,0,
+			0,1,0,0,
+			0,0,1,0,
+			0,0,0,1)
+
+	if(!C.parallax_offset.len)
+		C.parallax_offset["horizontal"] = 0
+		C.parallax_offset["vertical"] = 0
+
+/datum/hud/proc/update_parallax3()
+	var/client/C = mymob.client
+	//ACTUALLY MOVING THE PARALLAX
+	var/turf/posobj = get_turf(C.eye)
+
+	if(!C.previous_turf || (C.previous_turf.z != posobj.z))
+		C.previous_turf = posobj
+
+	//Doing it this way prevents parallax layers from "jumping" when you change Z-Levels.
+	C.parallax_offset["horizontal"] += posobj.x - C.previous_turf.x
+	C.parallax_offset["vertical"] += posobj.y - C.previous_turf.y
+
+	C.previous_turf = posobj
+
+	for(var/obj/screen/parallax/bgobj in C.parallax)
```

Comment:
So your results of profiling is meaningless. I don't have 50 people to test with and you know this,